src/microbe_stage/Microbe.cs

```diff
@@ -789,14 +789,41 @@ public void Divide()
         copyEntity.Compounds.ClearCompounds();
 
         var keys = new List<Compound>(Compounds.Compounds.Keys);
+        var reproductionCompounds = copyEntity.CalculateTotalCompounds();
 
-        // Split the compounds evenly between the two cells.
+        // Split the compounds between the two cells.
         foreach (var compound in keys)
         {
             var amount = Compounds.GetCompoundAmount(compound);
 
-            if (amount > 0)
+            if (amount <= 0)
+                continue;
+
+            // If the compound is for reproduction we give player and NPC microbes different amounts.
+            if (reproductionCompounds.TryGetValue(compound, out float divideAmount))
+            {
+                // Player cells receive the normal half split, NPC cells may receive less.
+                float amountToTake = amount * 0.5f;
+
+                if (!IsPlayerMicrobe)
+                {
+                    // NPC cells are capped at 90% of the amount of compound required to divide.
+                    amountToTake = Math.Max(amountToTake, amount - (divideAmount * 0.9f));
+                }
+
+                Compounds.TakeCompound(compound, amountToTake);
+
+                float amountToGive = Math.Min(amount * 0.5f, divideAmount * 0.9f);
```

Comment:
Shouldn't this be the other way around?
Because this method runs in the player cell.
So when the case is that `IsPlayerCell` then we need to cap the amountToTake from that cell, and then when we do that the amount to give should be equal to amountToTake. Something like:

```
if (!IsPlayerMicrobe)
{
    amountToTake = Math.Min(amountToTake, divideAmount * 0.9f);
}

Compounds.TakeCompound(compound, amountToTake);
var didntFit = copyEntity.Compounds.AddCompound(compound, amountToTake);
```

Reply:
The functionality in-game seemed fine but I can't wrap my head around the logic which way around the math should be to confirm that the amount taken and amount given to the other cell are equal, and that the non player cell won't end up with enough to reproduce.

Reply:
How it works right now is if the parent cell is a player they get 1/2 the compounds, always. If its an NPC cell they either get half taken away or 90% of divideAmount, whichever is larger. The child cell is always an NPC cell so it gets the same treatment but reversed, adding instead of taking away.

Reply:
With that explanation I can see that. Could you add like 2-3 clarifying comments in the code as that logic is pretty hard to unpack otherwise?

Reply:
Is that better?